dev/io.openliberty.accesslists.internal/src/com/ibm/ws/tcpchannel/internal/AccessLists.java

```diff
@@ -109,86 +115,100 @@ protected static AccessLists getInstance(AccessListKeysFacade keys) {
     /**
      * Query whether a given client address is denied by this configuration.
      * 
-     * @param remoteAddr
+     * @param addr
      * @return boolean
      */
-    public boolean accessDenied(InetAddress remoteAddr) {
-
-        String hostname = null; //F184719
-
-        // check the inclusion lists first to see if the client matches
-        if (includeAccess.getActive() || includeAccessNames.getActive()) {
-            boolean closeSocket = true;
-
-            if (includeAccess.getActive()) {
-                if (remoteAddr instanceof Inet6Address) {
-                    if (includeAccess.findInList6(remoteAddr.getAddress())) {
-                        closeSocket = false;
-                    }
-                } else {
-                    if (includeAccess.findInList(remoteAddr.getAddress())) {
-                        closeSocket = false;
-                    }
-                }
-            }
+    public boolean accessDenied(InetAddress addr) {
+        return nonEmptyIncludesAndNotFound(addr) || inExcludes(addr);
+    }
 
-            if (closeSocket && includeAccessNames.getActive()) {
-                // look at hostnames to check inclusion
+    /**
+     * We override the parent for TCP to add in hostname access lists
+     * 
+     * @param addr
+     */
+    @Override
+    protected boolean nonEmptyIncludesAndNotFound(InetAddress addr) {
 
-                hostname = remoteAddr.getHostName();
-                if (caseInsensitiveHostnames && (hostname != null)) {
-                    hostname = hostname.toLowerCase();
-                }
+        if (allIncludesEmpty()) {
```

Comment:
similar to my comment in GenericAccessLists, is there a reason for this helper `allIncludesEmpty()` since the `isActive()` checks will be duplicated in `inHostnameIncludes()` and `inAddressIncludes()`?

Reply:
I wanted to split the host and address list processing apart so that the address code could be common.
In that case, the problem with relying on the isActive check inside the hostname and address includes list logic is that, within the scope of ONE of those methods - neither can say definitively that something is excluded
(as omission always has to be from both if either is non-empty for that to happen.)

I don't want the host or address logic to refer to other's flag. 
So, if either handles both emptiness and inclusion (so that emptiness does not have to be tested
elsewhere) and returns false to mean denied is false - was it because the list was empty (exclusion still possible due to the other include list) or because it found the address on the list.

Of course I could 'invert' the logic to be 'not excluded because of my include list' (EmptyOrFound) 
but then EmptyOrFound(Hosts) || EmptyOrFound(Address) is not correct due to 'lazy OR'
and  neither is EmptyOrFound(Hosts) && EmptyOrFound(Address) nor XOR - basically I'd still have
to differentiate the boolean result between {Empty, Found, NotFound} outside the common code using isActive()

Reply:
Is this OK @wtlucy ?

Reply:
Thank you for the explanation @hutchig! I apologize for not catching this originally, this makes sense to me now, and I agree it's more elegant than the alternative - please disregard my original comment.